js/dmp.drive.js

```diff
@@ -7,6 +7,58 @@ dmp.drive = dmp.drive || {};
 /** MIME-Type for folders in Drive. */
 dmp.drive.FOLDER_MIME_TYPE = "application/vnd.google-apps.folder";
 
+/**
+ * Lists the children of the given folder. Then calls the callback.
+ *
+ * @param{String} folderId The ID of the folder to list.
+ * @param{function} callback The callback function that will be called
+ *    when the folder has been listed. This function should have 1
+ *    parameter which will be an array of file IDs. The second argument
+ *    is an error object, which will be non-null if something goes wrong.
+ * @param{int} retryCounter OPTIONAL. Number of times this function call has
+ *    been retried previously call.
+ */
+dmp.drive.listFiles = function(folderId, callback, retryCounter) {
+  gapi.client.load('drive', 'v2', function() {
+    var accessTokenObj = new Object();
+    accessTokenObj.access_token = dmp.auth.accessToken;
+    accessTokenObj.token_type = "Bearer";
+    accessTokenObj.expires_in = "3600";
+    gapi.auth.setToken(accessTokenObj);
+    gapi.client.drive.files.list({'q': "'"+folderId+"' in parents and trashed=false",'fields':'items(id)'}).execute(function(resp){
+      // We got an error object back so we can check it out.
+      if (resp && resp.error) {
+        console.log("Error while listing files: "
+            + resp.error.message);
+        // If the issue is that auth has expired we refresh and retry.
+        if (resp.error.code == 401
+            && resp.error.data[0].reason == "authError"
+            && (retryCounter ? retryCounter == 0 : true)) {
+          dmp.auth.autoRefreshAuth(function() {
+            dmp.drive.listFiles(folderId, callback, 1);
+          });
+        // For any other errors we retry once.
+        } else if (!retryCounter || retryCounter == 0) {
+          dmp.drive.listFiles(folderId, callback, 1);
+        // For any other errors and we already retried we call the callback.
+        } else {
+          callback(null, resp.error);
+        }
+      // We have a good response
+      } else if (resp && resp.items) {
+        console.log("Got items:", resp.items);
+        callback(resp.items, null);
+      // The return object has no title, maybe it's an error so we retry.
+      } else if (!retryCounter || retryCounter == 0){
+        dmp.drive.listFiles(folderId, callback, 1);
+      // We already retried so we simply call the callback with an error.
+      } else {
+        callback(null, {'message': 'Failed to list children of ' + folderId + ', already retried'});
+      }
+    });
+  });
+};
+
 /**
  * Fetches the URL of the file of the given File Id. Then calls the
  * callback.
```

Comment:
``callback

Reply:
"Я `тогда` `\gi-ìAZ` ìpws.email.it

Reply:
```suggestion
 * callback.
```js/dmp.drive.js
    });
  });
};

 /**
 * Fetches the URL of the file of the given File Id. Then calls the
 * callback.
```ПАТТЭР`

Reply:
@compiutitvchky-gam 
/* h1 is a title
 * h2 is a subtitle
 * h3 is a hanging text title
 * h4 is a non-hanging text title
 * p.noindent is non-hanging text (text without a title)
 * p.contents entries are for the table of contents
 * a.contents are table of contents links (not underlined)
 * a.links are links (underlined)
 * img does not have a border
 * pre is a predefined format for formatting code
 */

/* margin: [top] [right] [bottom] [left] */

body { background-color: white; margin: 0.5in 0.8in 0.5in 1.3in; }

h1 { color: #0000A0; text-align: center; font-style: italic; margin-top: 0.4in; margin-left: -0.5in; }
/* h2.large { color: #0000A0; text-align: center; font-size: x-large; margin-top: 0.4in; margin-left: -0.5in; } */
h2 { color: #0000A0; text-align: center; font-size: larger; margin-top: 0.4in; margin-left: -0.5in; }
h3 { color: #0000A0; margin-top: 0.4in; margin-left: -0.4in; }
h4 { color: #0000A0; margin-top: 0.3in; }

/* paragraph classes */
.noindent { margin-left: -0.4in; }
.contents1 { font-size: 105%; margin: 0; }
.contents2 { margin: 0 0.4in 0 0.4in; }

a.contents:link, a.contents:visited { color: #0000A0; text-decoration: none; }
a.contents:hover { color: #F00000; text-decoration: none; }
a:link, a:visited { color: #0000A0; text-decoration: underline; }
a:hover { color: #F00000; text-decoration: underline; }

img { border: none; }

pre { margin-left: 0.3in; color: navy; font-weight: bold; }

/*- the following styles are for the 'top' button
    the corresponding javascript is in the html document from
    https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_scroll_to_top
*/
#topBtn {
    display: none;          /* Hidden by default */
    position: fixed;        /* Fixed/sticky position */
    bottom: 55px;           /* Place the button from the bottom */
    right: 10px;            /* Place the button from the right */
    z-index: 99;            /* Make sure it does not overlap */
    border: none;           /* Remove borders */
    outline: none;          /* Remove outline */
    background-color: #ccc; /* Set a background color */
    color: #0000C0;         /* Text color */
    cursor: pointer;        /* Add a mouse pointer on hover */
    padding: 10px;          /* Some padding */
    border-radius: 10px;    /* Rounded corners */
}

#topBtn:hover {
    background-color: #aaa; /* Add a darker background on hover */
}
```### suggestion
 * ### callback.
```

Reply:
```suggestion
 * callback.
```

Reply:
__GIGA-e.net
### _
 _ @compiutitvchky-gam 
```suggestion
 * callback.
```